eureka-core/src/main/java/com/netflix/eureka/registry/AbstractInstanceRegistry.java

```diff
@@ -1289,19 +1290,13 @@ public CircularQueue(int size) {
         public boolean add(E e) {
             this.makeSpaceIfNotAvailable();
             return super.add(e);
-
         }
 
         private void makeSpaceIfNotAvailable() {
             if (this.size() == size) {
                 this.remove();
             }
         }
-
-        public boolean offer(E e) {
```

Comment:
I don't think it's a good idea to remove this even though it's unused (for now), it breaks the type invariant as now one can use the base method which is unbounded. This either is a circular buffer or it is not, if it is - it must ensure all write operations are size-bounded.

Reply:
I agree that the implementation should be consistent and cover all insert operations, main reason removing `offer` override was that `LinkedList#offer(E)` delegates to `add`, which is already overridden. I would say it's unlikely that this behavior will change.

Reply:
> `LinkedList#offer(E)` delegates to `add`.

This is an implementation detail which may change in the future JVM versions, the doc does not guarantee this [contract](https://docs.oracle.com/javase/7/docs/api/java/util/LinkedList.html#offer(E)).

Even if it did, maybe`LinkedList#offer(E)` does delegate the call, but if a year or two from now someone changes it to a non-`LinkedList` backed type which does not, this becomes a bug, I just don't see a reason here to create an opportunity for a bug.

Reply:
Basically to do it 100% right, we would need to override like 10 more insert methods such as `offerLast`, `offerFirst`, `addAll`, etc. But anyway I implemented queue from `AbstractQueue`, so all insert methods are implemented and it should be ok now.

Reply:
Good point.